diff --git a/rl-starter-files/scripts/train.py b/rl-starter-files/scripts/train.py
index 6b1b0c6..5a1de02 100644
--- a/rl-starter-files/scripts/train.py
+++ b/rl-starter-files/scripts/train.py
@@ -19,6 +19,10 @@ sys.path.insert(0, os.path.abspath('/Users/corinacaraconcea/Documents/UCL DSML/M
 import a2c
 import ppo
 
+from Minigrid.minigrid.__init__ import register_minigrid_envs
+
+register_minigrid_envs()
+
 # Parse arguments
 
 #The argparse module is a standard Python library for writing user-friendly command-line interfaces.
diff --git a/rl-starter-files/utils/env.py b/rl-starter-files/utils/env.py
index fa4f36d..24b9b4b 100644
--- a/rl-starter-files/utils/env.py
+++ b/rl-starter-files/utils/env.py
@@ -5,3 +5,4 @@ def make_env(env_key, seed=None, render_mode=None):
     env = gym.make(env_key, render_mode=render_mode)
     env.reset(seed=seed)
     return env
+
